src/libexpr/primops.cc

```diff
@@ -1734,8 +1734,12 @@ static void prim_match(EvalState & state, const Pos & pos, Value * * args, Value
                 mkString(*(v.listElems()[i] = state.allocValue()), match[i + 1].str().c_str());
         }
 
-    } catch (std::regex_error &) {
-        throw EvalError("invalid regular expression ‘%s’, at %s", re, pos);
+    } catch (std::regex_error &e) {
+        if (e.code() == std::regex_constants::error_space) {
+          throw EvalError("regular expression exceeds _GLIBCXX_REGEX_STATE_LIMIT ‘%s’, at %s", re, pos);
```

Comment:
Just a nitpick: `_GLIBCXX_REGEX_STATE_LIMIT` is libstdc++ specific.  Maybe something like "insufficient memory for regular expression ..."?